cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/CommandUtil.java

```diff
@@ -115,11 +155,124 @@ public static void initProject(Path path) throws IOException {
+        // - Ballerina.toml
+        // - Package.md
+        // - Module.md
+        // - main.bal
+        // - resources
+        // - tests
+        //      - main_test.bal
+        //      - resources/
+        // - .gitignore       <- git ignore file
+        initProject(path, packageName);
+        applyTemplate(path, template);
+        Path gitignore = path.resolve(ProjectConstants.GITIGNORE_FILE_NAME);
+
+        Files.createFile(gitignore);
 
+        String defaultGitignore = FileUtils.readFileAsString(NEW_CMD_DEFAULTS + File.separator + GITIGNORE);
+
+        Files.write(gitignore, defaultGitignore.getBytes(StandardCharsets.UTF_8));
     }
 
+    /**
+     * Get the list of templates.
+     *
+     * @return list of templates
+     */
+    public static List<String> getTemplates() {
+        try {
+            Path templateDir = getTemplatePath();
+            Stream<Path> walk = Files.walk(templateDir, 1);
+
+            List<String> templates = walk.filter(Files::isDirectory)
+                    .filter(directory -> !templateDir.equals(directory))
+                    .filter(directory -> directory.getFileName() != null)
+                    .map(directory -> directory.getFileName())
+                    .map(fileName -> fileName.toString())
+                    .collect(Collectors.toList());
+
+            if (null != jarFs) {
+                return templates.stream().map(t -> t
+                        .replace(jarFs.getSeparator(), ""))
+                        .collect(Collectors.toList());
+            } else {
+                return templates;
+            }
+
+        } catch (IOException | URISyntaxException e) {
+            // we will return an empty list if error.
+            return new ArrayList<String>();
+        }
+    }
+
+    /**
+     * Get the path to the given template.
+     *
+     * @return path of the given template
+     * @throws URISyntaxException if any URISyntaxException occured
+     */
+    private static Path getTemplatePath() throws URISyntaxException {
+        URI uri = CommandUtil.class.getClassLoader().getResource(CREATE_CMD_TEMPLATES).toURI();
+        if (uri.toString().contains("!")) {
+            final String[] array = uri.toString().split("!");
+            return jarFs.getPath(array[1]);
+        } else {
+            return Paths.get(uri);
+        }
+    }
+
+    /**
+     * Apply the template to the created module.
+     *
+     * @param modulePath path to the module
+     * @param template template name
+     * @throws IOException if any IOException occurred
+     * @throws URISyntaxException if any URISyntaxException occurred
+     */
+    public static void applyTemplate(Path modulePath, String template) throws IOException, URISyntaxException {
+        Path templateDir = getTemplatePath().resolve(template);
+
+        try {
+            Files.walkFileTree(templateDir, new FileUtils.Copy(templateDir, modulePath));
```

Comment:
you dont need catch and re-throw the same exception here. Having the `throws` in the method signature would suffice